src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private float threshold;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        int index = indexBucket(key);
+        Node<K, V> node = table[index];
+        Node<K, V> newNode = new Node<>(key, value);
+        if (node == null) {
+            table[index] = newNode;
+        } else {
+            // в цьому бакеті щось є(пробігтись, перевірячи ключі,
+            // якщо == міняємо, якщо колізія - некст
+            Node<K, V> prev = node;
+            while (node != null) {
+                if (key == node.key || (key != null && key.equals(node.key))) {
+                    node.value = value;
+                    return;
+                }
+                prev = node;//1  2  3 null
+                node = node.next;
+            }
+            prev.next = newNode;
+        }
+        size++;
+        // пошук потрібного бакету
+        // чи бакет емпті, якщо так - шукаємо до кінця списку
     }
 
     @Override
     public V getValue(K key) {
+        int index = indexBucket(key);
```

Comment:
```suggestion
        int index = getIndexBucket(key);
```
use verbs for methods names